game/player.go

```diff
@@ -310,12 +312,23 @@ func (p *Player) appendActionsForInstant(answer []*Action, card *Card) []*Action
 								for _, index := range c {
 									selected = append(selected, p.game.Lands()[index].Id)
 								}
-								answer = append(answer, &Action{
+								action := &Action{
 									Type:     Play,
 									Card:     card,
 									Selected: selected,
 									Target:   targetCreature.Id,
-								})
+								}
+
+								gJson := p.game.serialized()
```

Comment:
doing all this serialization inside `appendActionsForInstant` seems like it would be really slow. is this worth avoiding some duplicated state? I think you would basically never want methods inside action-generation to do a ton of serialization.